Rf

Pratichhya opened the pull request:
Tutorial for LPS

Pratichhya commented:
> * get_cloud_free_pixels -> should we use to_scl_dilation_mask instead?

I assume you mean in this manner:

```
scl.process(
        "to_scl_dilation_mask",
        data = scl,
        kernel1_size=17, kernel2_size=77,
        mask1_values=[2, 4, 5, 6, 7],
        mask2_values=[3, 8, 9, 10, 11],
        erosion_kernel_size=3
        )
```

But with this function, I want to do something similar to the best available pixel. And then apply certain thresholding on it. Also, with this function, I wanted to introduce the possibility to define the reducer such that in other cases, the user can choose which image they want to train on (earlier, at the time or later after the event). But I can give it a try, if not, considering the option of reducer and sticking to the standard kernel size of `to_scl_dilation_mask` without any thresholding still gives a nice result. 😃 

> * can we try to move the rename_labels inside the udf ad apply_metadata? Should work on applying neighbourhoo functions

Sure, I assume you meant in similar manner: https://open-eo.github.io/openeo-python-client/udf.html#examples

> * renaming of `combined` to `merged_cube`, as a new openeo process called combine_cubes might be introduced in the future

Not sure how you meant here.


> * discussion: promote fit_class_random_forest or rather extractions + offline training as in practice this is more used

Do you mean providing additional text supporting the use of `fit_class_random_forest ` ?

> * MlModel.load_ml_model can be invoced on a connection directly (removes an import and follows the standard notation of openeo better imo)

Could you direct me or provide me with an example snippet?
 
@VictorVerhaert thank you for all the feedback, for those that are not replied in my comment. I agree to your suggestion and will try to address them all.

Pratichhya commented:
@VictorVerhaert @HansVRP thank you again for the feedback, I tried to address all of them. Please let me know if you have additional suggestions.

HansVRP commented:
instead of dumping eerything in helper; you could also have an eo_extractor.py with your s1/s2 pipeline


I would rehting the naming of the notebooks; RF_forest pire and inference, do not really give a sense of order: 

1. Random Forest Model Training
2. Random Forest Model Inference

##1
Tenerife(Canary Islands) is missing a space.

available by the ecosystem --> available in the CDSE?

 Sentinel-2 provides high-resolution multispectral optical imagery, which is essential for extracting spectral bands, vegetation indices, and textural features, including Gray-Level Co-occurrence Matrix (GLCM) textures derived from the short-wave infrared bands
--> I would leave the GLCM explanation out here; its not tied to sentinel2, but a method to be describe in your UDF.

 TA temporal reduction
--> unclear acronym TA?

# Merge the two cubes
training_cube = s2_feature_cube.merge_cubes(s1_feature_cube)
--> plot the process graph here

Unclear why [16-18] are seperate cells




#2
In openEO, users can chain together processing steps in a process graph. openEO allows processes to be chained together in a process graph to build a particular algorithm. Often, users have specific (sub)graphs that reoccur in the same process graph or even in different process graphs or algorithms.  These process graphs can be reused at various scales and with varying inputs, making them highly flexible and efficient for repeated analysis.

is very repetitive

The additional required parameter is public_url. If the process is saved as a UDP on the back-end, this URL will be automatically generated and made publicly accessib if flagged as publicle
--> typo or renders odd

 hosted JSON file—is generally recommendt.
--> typo

Here I miss some context. I believe this is very relevant that as a researcher you can train your model; and share the data extraction+model inference pipeline for external users. 
Often on platforms like hugging face you can find the model back and if you are lucky code on how the data was preprocessed. 

Here is really low effort for anyone to use your workflow.

Pratichhya commented:
@HansVRP thank you for a detailed check.

I addressed all of them. 

> Here is really low effort for anyone to use your workflow.

For this I updated the final paragraph with a bit more description.

VictorVerhaert reviewed (approved):
approving but left some comments to be addressed
